### ISSUE TITLE
Keeps crashing

### ISSUE THREAD

**Issue by tswaters** [glen-84/vscode-sass-lint#4]
I keep getting the following flash shown:

> The Sass Lint server crashed 5 times in the last 3 minutes. The server will not be restarted.

I can't see any stack trace or anything but the vscode instance must be restarted to get warnings and errors from this extension again.

This is on Windows 10.  
sass-lint: v.1.9.0
node: v6.3.0
vscode: 1.4.0
there are 4 sass files in the project.

It will happen as I'm making changes to the sass files attempting to fix up the problems.

My config file:

``` yml
files:
  include: 'assets/sass/*.s+(a|c)ss'
rules:
  property-sort-order:
  - 1
  -
    order: 'smacss'
```

And relevant settings in vscode settings:

``` json
{
  "scss.validate": false,
  "sasslint.enable": true,
  "sasslint.configFile": ".sass-lint.yml"
}
```

I wish I could be of more help - do you know of a way I can get the stack trace from the language service?  The only thing the vscode dev tools report is a console.error in messageService.js.

**Comment by glen-84** [comment]
Hi Tyler,
1. Does sass-lint work from the command line?
2. Does it also crash with sass-lint 1.8.2? (I haven't tried 1.9.0 yet)

> do you know of a way I can get the stack trace from the language service

I'm not sure, no. I assume there was nothing in the output window for extensions?

**Comment by tswaters** [comment]
Yes it works from command line.  It also works from VSCode initially;  I am able to see the warnings/errors show up up until I see the flash that it has crashed too often and then no warnings/errors show up.

The only thing to the console I get relates to the actual flash being shown.

It should be noted I have a watch task running that uses node-sass that compiles the files to a different directory... could it possibly relate to locking issues?

I'll give 1.8.2 a try when I get home from work today.

**Comment by glen-84** [comment]
> could it possibly relate to locking issues?

Seems unlikely. I also use node-sass (via gulp-sass). It's easy to test though (see if the problem persists while the watch task is not running).

**Comment by glen-84** [comment]
@tswaters Any luck?

It's quite possibly caused by [this issue](https://github.com/tonyganch/gonzales-pe/issues/189).

**Comment by tswaters** [comment]
Ah yes forgot about this, sorry about that.

I tried it under 1.8.2 and it doesn't seem to crash.   However, when it was crashing it was a bit spotty and would typically happen "eventually".  I've been playing with it for a few minutes now and I haven't seen a crash yet, but I'm not entirely convinced downgrading the version fixed it.

I looked through the changes between those two versions and the major thing that stands out was gonzales-pe getting updating.... the actual API however wasn't changed at all.  sass-lint still swallows errors from gonzales-pe, and will still return the same `fatal` message if anything blows up.

I'm going to close this for now and will reopen if I can figure a way to recreate it reliably.  

FWIW, I encountered this working on the sass files here:  https://github.com/tswaters/klondike/tree/master/assets/sass  -- specifically the funky nth-child bits at the bottom of _cards.scss

I was trying to fix up 'force-*-nesting' failures (this is toggled off in sass-lint.yml now) -- once I fixed those up, it went to 'nesting too deep' ... my guess would be while I was changing this, some combination of text no one expects was run through the linter and blew everything up.... 5 times.
